src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -86,9 +83,16 @@ class CoreWorkerDirectTaskSubmitter {
 
   /// Request a new worker from the raylet if no such requests are currently in
   /// flight.
-  void RequestNewWorkerIfNeeded(const TaskSpecification &resource_spec)
+  void RequestNewWorkerIfNeeded(const TaskSpecification &resource_spec, const rpc::Address *address = nullptr)
       EXCLUSIVE_LOCKS_REQUIRED(mu_);
 
+  /// Callback for when the raylet grants us a worker lease. The worker is returned
+  /// to the raylet once it finishes its task and either the lease term has
+  /// expired, or there is no more work it can take on.
+  ///
+  /// \param[in] addr The (addr, port) pair identifying the worker.
+  void HandleWorkerLeaseGranted(const WorkerAddress addr);
```

Comment:
You probably know this but this will need to update test_direct_task_transport

Reply:
Fixed and added a test for spillback.